### PR TITLE
fix: `table.keep_only_columns` now maps column names to correct data

### LINKED ISSUE #115: `table.keep_only_columns` maps column names to wrong data
### Describe the bug

The method `keep_only_columns` in the `Table` class returns a table with the correct columns and the correct data. However, if the order of the columns names differs between the table and the `column_names` argument, the column names get mapped to the wrong data: The columns names of the resulting table are still in the same order as in the original table. The data, however, is ordered like the `column_names` list.

### To Reproduce

```py
from safeds.data.tabular.containers import Column, Table

table = Table.from_columns(
    [
        Column("A", [1, 4]),
        Column("B", [2, 5]),
        Column("C", [3, 6]),
    ]
)

print(table.keep_only_columns(["C", "A"]))
```

**Result:**

```
   A  C
0  3  1
1  6  4
```

### Expected behavior

```
   C  A
0  3  1
1  6  4
```

Columns should be ordered as specified by the `column_names` argument.

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by Marsmaennchen221**
Closes #115 .

### Summary of Changes

Fixed `table.keep_only_columns` to rearrange the column names correctly
Fixed grammar in two docs

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.1) status: ✅ [SUCCESS](https://github.com/Safe-DS/Stdlib/actions/runs/4714677915)

| Descriptor  |                                 Linter                                 |Files|Fixed|Errors|Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.1/descriptors/python_black)          |    2|    0|     0|0.72s       |
|✅ PYTHON    |[mypy](https://megalinter.io/6.22.1/descriptors/python_mypy)            |    2|     |     0|1.55s       |
|✅ PYTHON    |[ruff](https://megalinter.io/6.22.1/descriptors/python_ruff)            |    2|    0|     0|0.04s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.1/descriptors/repository_git_diff)|yes  |     |no    |0.02s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4714677915)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/tabular/containers/_table.py [hunk 1]
```diff
@@ -578,7 +578,7 @@ def keep_only_columns(self, column_names: list[str]) -> Table:
         if len(invalid_columns) != 0:
             raise UnknownColumnNameError(invalid_columns)
         transformed_data = self._data[column_indices]
-        transformed_data.columns = [name for name in self._schema.get_column_names() if name in column_names]
+        transformed_data.columns = [self._schema.get_column_names()[i] for i in column_indices]
```
Comment (lars-reimann) [inline src/safeds/data/tabular/containers/_table.py:581]:
```suggestion
        transformed_data.columns = column_names
```

Isn't this sufficient?

File: tests/safeds/data/tabular/containers/_table/test_keep_only_columns.py [hunk 2]
```diff
@@ -12,6 +12,15 @@ def test_keep_columns() -> None:
     assert not transformed_table.schema.has_column("B")
 
 
+def test_keep_columns_order() -> None:
```
Comment (lars-reimann) [inline tests/safeds/data/tabular/containers/_table/test_keep_only_columns.py:15]:
```suggestion
def test_keep_only_columns_order() -> None:
```

File: tests/safeds/data/tabular/containers/_table/test_keep_only_columns.py [hunk 3]
```diff
@@ -12,6 +12,15 @@ def test_keep_columns() -> None:
...
+    table = Table.from_csv_file(resolve_resource_path("test_table_from_csv_file.csv"))
+    transformed_table = table.keep_only_columns(["B", "A"])
+    assert table.to_columns()[0] == transformed_table.to_columns()[1]
+    assert table.to_columns()[1] == transformed_table.to_columns()[0]
+    assert table.get_column("A") == transformed_table.get_column("A")
+    assert table.get_column("B") == transformed_table.get_column("B")
+
+
 def test_keep_columns_warning() -> None:
```
Comment (lars-reimann) [inline tests/safeds/data/tabular/containers/_table/test_keep_only_columns.py:24]:
```suggestion
def test_keep_only_columns_warning() -> None:
```

File: tests/safeds/data/tabular/containers/_table/test_keep_only_columns.py
Comment (lars-reimann) [inline tests/safeds/data/tabular/containers/_table/test_keep_only_columns.py:1]: While we're at it, `test_keep_columns` should also be called `test_keep_only_columns` instead.

**Review by lars-reimann — CHANGES_REQUESTED** [review]
The title of the PR should also be changed, so a [good changelog entry](https://stdlib.safe-ds.com/en/stable/CHANGELOG/) can be created: 

* It's not necessary to link to the issue in the title, that's what the `Closes #115` is for. 
* The title should be something that a user can understand
* It's not necessary to point out in the title that tests were added that reproduce the bug. That is implied.

It could be something like "fix: `table.keep_only_columns` now maps column names to correct data"

File: src/safeds/data/tabular/containers/_table.py (hunk 1 above)
Comment (Marsmaennchen221) [reply src/safeds/data/tabular/containers/_table.py:581]: yes you are right

**Review by lars-reimann — APPROVED** [review]
LGTM, thanks!

**Pull request merged**